File: WorkManagerApp/app/src/main/java/ir/hajkarami/workmanagerapp/MainActivity.java
package ir.hajkarami.workmanagerapp;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.work.OneTimeWorkRequest;
import androidx.work.WorkManager;
import androidx.work.WorkRequest;

public class MainActivity extends AppCompatActivity {

    private Button btn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        btn = findViewById(R.id.button);

        // Making use of worker
        WorkRequest w = new OneTimeWorkRequest.Builder(MyWorker.class).build();
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                WorkManager.getInstance(getApplicationContext()).enqueue(w);
            }
        });

    }
}

File: DataBinding/app/src/main/java/ir/hajkarami/databinding/MainActivity.kt
package ir.hajkarami.databinding

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.databinding.DataBindingUtil
import ir.hajkarami.databinding.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var activityMainBinding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        activityMainBinding = DataBindingUtil.setContentView(this,R.layout.activity_main)

        val p1 = Person("Saleh","mshajkrami@gmail.com")


        activityMainBinding.person = ObservablePerson(p1)
        val clickHandler = MyClickHandler(this)
        activityMainBinding.clickHandler = clickHandler


    }
}